### PR TITLE
Fix `:log-open` when `--log` is specified

### PR DESCRIPTION
**Pull request by alevinval**
The command always opened the default log file path. The path can be customized passing the `--log` optional argument. When this is the case the `open-log` command should respect it.

### CODE REVIEW

**Review by pascalkuthe — DISMISSED** [review]
thanks!

**Review by the-mikedavis — COMMENTED** [review]
I think we should use the same strategy as the config file so we don't need to tote around this log_file parameter on more structs and constructors: https://github.com/helix-editor/helix/blob/618620b36907bb2cbe29cd0e737efad9f651a00d/helix-loader/src/lib.rs#L12-L27

https://github.com/helix-editor/helix/blob/618620b36907bb2cbe29cd0e737efad9f651a00d/helix-loader/src/lib.rs#L124-L129

**Comment by alevinval** [comment]
@the-mikedavis really nice! much cleaner, I've also done some boy-scouting around that code, let me know if you are cool with it.

**Review by the-mikedavis — COMMENTED** [review]
I believe we also need an `ensure_log_dir` since that may be in a different directory than config (by deafult it is for example). I think it would be fine for those functions to be called in the initialize_ functions since those are only meant to be called once at initialization anyways

File: helix-loader/src/lib.rs [hunk 1]
```diff
@@ -122,10 +121,11 @@ pub fn cache_dir() -> PathBuf {
...
 pub fn config_file() -> PathBuf {
-    CONFIG_FILE
-        .get()
-        .map(|path| path.to_path_buf())
-        .unwrap_or_else(|| config_dir().join("config.toml"))
+    CONFIG_FILE.get().map(|path| path.to_path_buf()).unwrap()
+}
+
+pub fn log_file() -> PathBuf {
+    LOG_FILE.get().map(|path| path.to_path_buf()).unwrap()
```
Comment (alevinval) [inline helix-loader/src/lib.rs:128]: I was about to write the following here (and the equivalent for the `config_file()`)

```rust
unwrap_or_else(|| {
   initialize_log_file(None)
   log_file()
})
```

But then I really think we should `unwrap()` and potentially fail. A failure here means we have seriously broken the initialization steps in the `main.rs`, it seems less misleading than re-replicating the initialization of the cell in two places.

**Review by the-mikedavis — APPROVED** [review]

**Review by pascalkuthe — APPROVED** [review]

**Pull request merged**